pkg/processor/runtime/dotnetcore/Wrapper.cs

```diff
@@ -12,31 +12,90 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
 using Nuclio.Sdk;
+using System;
 
 namespace processor
 {
 
     public class Wrapper
     {
         private delegate object MethodDelegate(Context context, Event eve);
+        private delegate void InitContextDelegate(Context context);
         private static MethodDelegate methodDelegate;
+        private static string initContextFunctionName = "InitContext";
+        private Type methodType;
 
         private ISocketHandler socketHandler;
+        private Context context;
 
         public Wrapper(string dllPath, string typeName, string methodName, string socketPath)
         {
+            
             CreateTypeAndFunction(dllPath, typeName, methodName);
-            StartUnixSocketHandler(socketPath);
+
+            InitUnixSocketHandler(socketPath);
+
+            //Run the function's method that initializes the context
+            try
+            {
+                ExecuteInitContext();
```

Comment:
you essetnialy create and initialize the whole context, not just "initialize" the user-given function. 
perhaps would be better to
`Initialize()` - where you CreateTypeAndFunction, InitUnixSocketHandler and create the context object.
then you
`InitContext()` - where you execute (if given) the user's `InitContext(context)` function.